discovery-provider/src/tasks/index_user_bank.py

```diff
@@ -99,32 +100,83 @@ def get_tx_in_db(session: Session, tx_sig: str) -> bool:
     return exists
 
 
-def refresh_user_balance(session: Session, redis: Redis, user_bank_acct: str):
-    user_id: Optional[Tuple[int, None]] = (
-        session.query(User.user_id)
+def refresh_user_balances(session: Session, redis: Redis, accts=List[str]):
+    results = (
+        session.query(User.user_id, UserBankAccount.bank_account)
         .join(
             UserBankAccount,
             and_(
-                UserBankAccount.bank_account == user_bank_acct,
+                UserBankAccount.bank_account.in_(accts),
                 UserBankAccount.ethereum_address == User.wallet,
             ),
         )
         .filter(User.is_current == True)
-        .one_or_none()
+        .all()
     )
     # Only refresh if this is a known account within audius
-    if user_id:
-        logger.info(
-            f"index_user_bank.py | Refresh user_id = {user_id}, {user_bank_acct}"
-        )
-        enqueue_immediate_balance_refresh(redis, [user_id[0]])
+    if results:
+        user_ids = [user_id[0] for user_id in results]
+        logger.info(f"index_user_bank.py | Refresh user_ids = {user_ids}")
+        enqueue_immediate_balance_refresh(redis, user_ids)
+    return results
 
 
 create_token_account_instr: List[InstructionFormat] = [
     {"name": "eth_address", "type": SolanaInstructionType.EthereumAddress},
 ]
 
 
+def index_tip(
+    session, sender_account, receiver_account, user_id_accounts, meta, tx_sig, slot
+):
+    if user_id_accounts and len(user_id_accounts) == 2:
```

Comment:
Rather than nesting the happy path, would be cleaner to early return if != 2 imo

Reply:
I like returning early for errors etc but I think this nesting makes sense - basically `if tip -> index tip` vs `if not tip -> return -> else index tip`